### DISCUSSION TITLE
Text Classification with Naive Bayes

### DISCUSSION THREAD

**Discussion by naturallyfunny** [teamalgoritma/community#796] [Q&A] [answered]
Saya penasaran dan ingin memastikan, apa yang terjadi jika model naive bayes dimasukkan text yang terdapat huruf yang bukan merupakan prediktor modelnya?

**Comment by arkanayudhistira** [comment, marked as answer]
Dear @ArdiantheGreat,

Misalkan suatu model Naive Bayes (berlaku juga untuk model-model lain) telah ditrain menggunakan suatu dataset text yang sudah dijadikan Document Term Matrix (atau Tokenization sejenisnya) dengan prediktor A, B, C, dan D. 

Jika model tersebut ingin memprediksi data text baru yang berbeda prediktornya, misal A , B, C, E, dan F, maka harus dibuat menjadi sebuah DTM yang telah dicocokkan hingga sesuai dengan DTM milik data train, maka nantinya akan jadi memiliki nilai:

| A | B | C | D |
|---|---|---|---|
| 1 | 1 | 1 | 0 |

dimana prediktor E dan F akan diabaikan karena tidak ada pada data train. 
Berikut referensi untuk melakukan proses tersebut: https://stackoverflow.com/q/15751171

Semoga membantu!

**Reply by IrfanChairurrachman** [reply]
Halo @ArdiantheGreat , untuk menjaga agar forum diskusi berjalan kondusif, kami izin melakukan close pada diskusi ini karena belum ada respon setelah 7 hari respon terakhir. Semoga dapat dipahami. Terima kasih.
